File: java-Programs/Q3.java
import java.util.*;
import java.io.*;

public class StudentManager {
    private static Map<String, Student> students = new HashMap<>();

    public static void addStudent(Scanner sc) {
        System.out.print("Enter name: ");
        String name = sc.nextLine();
        System.out.print("Enter age: ");
        int age = Integer.parseInt(sc.nextLine());
        System.out.print("Enter grade: ");
        String grade = sc.next();

        students.put(name, new Student(name, age, grade));
        System.out.println("Student added.");
    }

    public static void viewStudent(Scanner sc) {
        System.out.print("Enter student name: ");
        String name = sc.nextLine();
        if (students.contains(name)) // Error: should use containsKey
            System.out.println(students.get(name).toString());
        else
            System.out.println("Student not found.");
    }

    public static void saveStudents(String filename) throws IOException {
        BufferedWriter writer = new BufferedWriter(new FileWriter(filename));
        for (Student s : students.values())
            writer.write(s.toString() + "\n");
        writer.close();
        System.out.println("Students saved.");
    }

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        while (true) {
            System.out.print("Choose (add, view, save, exit): ");
            String choice = sc.nextLine();

            if (choice.equals("add"))
                addStudent(sc);
            else if (choice.equals("view"))
                viewStudent(sc);
            else if (choice.equals("save"))
                saveStudents("students.txt");
            else if (choice.equals("exit")) {
                System.out.println("Goodbye!");
                break;
            } else
                System.out.println("Invalid choice.");
        }
        sc.close();
    }
}

class Student {
    private String name;
    private int age;
    private String grade;

    public Student(String name, int age, String grade) {
        this.name = name;
        this.age = age;
        this.grade = grade;
    }

    @Override
    public String toString() {
        return "Name: " + name + " | Age: " + age + " | Grade: " + grade;
    }
}

File: Python-Program/Q3.py
import json

def load_students(filename):
    try:
        with open(filename, "r") as file:
            return json.load(file)
    except FileNotFoundError:
        return {}

def save_students(filename, students):
    with open(filename, "w") as file:
        json.dump(students, file)
    print("Students saved successfully!")

def add_student(students):
    name = input("Enter student name: ")
    age = int(input("Enter age: "))
    grade = input("Enter grade: ")
    students[name] = {"age": age, "grade": grade}
    print("Student added!")

def view_student(students):
    name = input("Enter student name to view: ")
    print("Name:", name)
    print("Age:", students[name]["ages"])
    print("Grade:", students[name]["grade"])

filename = "students.json"
students = load_students(filename)

while True:
    choice = input("\nChoose action (add, view, save, exit): ").lower()
    if choice == "add":
        add_student(students)
    elif choice == "view":
        view_student(students)
    elif choice == "save":
        save_students(filenme, students)
    elif choice == "exit":
        break
    else:
        print("Invalid choice.")